common/primitives/src/messages.rs

```diff
@@ -0,0 +1,110 @@
+use crate::msa::MessageSenderId;
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+use sp_runtime::traits::AtLeast32BitUnsigned;
+use sp_std::{prelude::*, vec};
+
+pub type SchemaId = u16;
+
+#[derive(Default, Clone, Encode, Decode, PartialEq, Debug, TypeInfo, Eq)]
+pub struct Message<AccountId> {
```

Comment:
why put this in common?

Reply:
It is being used in `MessageResponse`. One thing I think might make sense is to include all of Message field into MessageResponse (instead of containing it) like a proper DTO and put the `Message` just into the pallet and not expose it.